src/tinuous/base.py

```diff
@@ -64,27 +64,41 @@ def from_travis_event(cls, travis_event: str) -> Optional["EventType"]:
 class APIClient:
     MAX_RETRIES = 10
 
-    def __init__(self, base_url: str, headers: Dict[str, str]):
+    def __init__(self, base_url: str, headers: Dict[str, str], is_github: bool = False):
         self.base_url = base_url
         self.session = requests.Session()
         self.session.headers.update(headers)
+        self.is_github = is_github
 
     def get(self, path: str, **kwargs: Any) -> requests.Response:
         if path.lower().startswith(("http://", "https://")):
             url = path
         else:
             url = self.base_url.rstrip("/") + "/" + path.lstrip("/")
         i = 0
-        r = self.session.get(url, **kwargs)
-        while r.status_code >= 500 and i < self.MAX_RETRIES:
-            log.warning(
-                "Request to %s returned %d; waiting & retrying", url, r.status_code
-            )
-            i += 1
-            sleep(i)
+        while True:
             r = self.session.get(url, **kwargs)
-        r.raise_for_status()
-        return r
+            if r.status_code >= 500 and i < self.MAX_RETRIES:
+                log.warning(
+                    "Request to %s returned %d; waiting & retrying", url, r.status_code
+                )
+                i += 1
+                sleep(i)
```

Comment:
This week result in up to about 50 sec sleep total with 10 retries. May be square it?

Reply:
Done.

Reply:
thanks!